### PR TITLE
discard reserved fields from channel.Update() channelData

### PR DESCRIPTION
**Pull request by gumuz**
# Submit a pull request

## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

### CODE REVIEW

File: src/channel.ts [hunk 1]
```diff
@@ -482,6 +482,12 @@ export class Channel<
     channelData: ChannelData<ChannelType>,
     updateMessage?: Message<AttachmentType, MessageType, UserType>,
   ) {
+    // Strip out reserved names that will result in API errors.
+    const reserved = ['config', 'cid', 'created_by', 'id', 'member_count', 'type'];
```
Comment (vishalnarkhede) [inline src/channel.ts:486]:
```suggestion
    const reserved = ['config', 'cid', 'created_by', 'id', 'member_count', 'type', 'created_at', 'updated_at', 'last_message_at'];
```

Comment (gumuz) [reply src/channel.ts]: thanks!

**Review by vishalnarkhede — APPROVED** [review]

**Review by gumuz — COMMENTED** [review]
Looks good to me! @nhannah

**Comment by veloware** [comment]
any idea of when this will be released? Thanks

**Comment by gumuz** [comment]
> any idea of when this will be released? Thanks

hey @veloware these changes should now be included in v2.5.0

**Comment by veloware** [comment]
just saw that, thanks!

**Pull request merged**